src/Database/LSMTree/Internal/Snapshot.hs

```diff
@@ -200,13 +200,15 @@ snapshotRuns ::
      (MonadMask m, MonadMVar m)
   => TempRegistry m
   -> NamedSnapshotDir
-  -> SnapLevels (Run m h)
+  -> SnapLevels (Ref (Run m h))
   -> m (SnapLevels RunNumber)
-snapshotRuns reg (NamedSnapshotDir targetDir) levels = for levels $ \run -> do
-    let sourcePaths = Run.runRunFsPaths run
-    let targetPaths = sourcePaths { runDir = targetDir }
-    hardLinkRunFiles reg (Run.runHasFS run) (Run.runHasBlockIO run) sourcePaths targetPaths
-    pure (runNumber targetPaths)
+snapshotRuns reg (NamedSnapshotDir targetDir) levels =
+    for levels $ \run@(DeRef Run.Run { Run.runHasFS = hfs,
+                                       Run.runHasBlockIO = hbio }) -> do
+      let sourcePaths = Run.runFsPaths run
+      let targetPaths = sourcePaths { runDir = targetDir }
+      hardLinkRunFiles reg hfs hbio sourcePaths targetPaths
+      pure (runNumber targetPaths)
```

Comment:
The diff would have been smaller if we only changed `... \run -> ...` to `... \(DeRef run) -> ...`

Reply:
Fair, but I'm not going to bother changing it now. :grin: